app/controllers/ReviewController.scala

```diff
@@ -17,19 +22,60 @@ class ReviewController @Inject()(
                                   configuration: Configuration)(
                                   implicit val ex: ExecutionContext) extends AbstractController(controllerComponents) with play.api.i18n.I18nSupport {
 
+  val form = Form(
+    mapping(
+      "confirmRecordTransfer" -> text
+        .verifying("Must answer yes", s => hasAgreed(s)),
+      "confirmOpen" -> text
+        .verifying("Must answer yes", s => hasAgreed(s)),
+      "confirmTnaOwnership" -> text
+        .verifying("Must answer yes", s => hasAgreed(s)),
+    )(ReviewData.apply)(ReviewData.unapply)
+  )
+
   def index(consignmentId: Int) = Action.async {
     implicit request =>
+      getDetailsToReview(consignmentId).map(reviewDetails =>
+        Ok(views.html.review(reviewDetails.consignment, reviewDetails.fileChecks, form))
+      )
+  }
+
+  def submit(consignmentId: Int) = Action.async { implicit request: Request[AnyContent] =>
+
+    val errorFunction: Form[ReviewData] => Future[Result] = { formWithErrors: Form[ReviewData] =>
+      getDetailsToReview(consignmentId).map(reviewDetails =>
+        BadRequest(views.html.review(reviewDetails.consignment, reviewDetails.fileChecks, formWithErrors))
+      )
+    }
+    val successFunction: ReviewData => Future[Result] = { formData: ReviewData =>
+      Future.apply(Redirect(routes.TransferConfirmationController.index()))
+    }
+
+    val formValidationResult: Form[ReviewData] = form.bindFromRequest
 
-      val appSyncClient = client.graphqlClient
```

Comment:
Probably should not use the val name "appSyncClient" as no longer using app sync.

Reply:
Good point. I've renamed them all. :+1: